src/main/java/uk/gov/hmcts/reform/em/orchestrator/automatedbundling/AutomatedCaseUpdater.java

```diff
@@ -43,37 +58,78 @@ public AutomatedCaseUpdater(ConfigurationLoader configurationLoader,
-
-        BundleConfiguration configuration = configurationLoader.load(configurationName);
-        final ArrayNode bundles = ccdCallbackDto
-            .findCaseProperty(ArrayNode.class)
-            .orElseGet(() -> {
+
+        List<String> bundleConfigurations = prepareBundleConfigs(ccdCallbackDto);
+
+        List<CcdBundleDTO>  ccdBundleDtos = populateBundleConfigs(ccdCallbackDto, bundleConfigurations);
+
+        //Make call for Stitching only after the validation is completed for the Bundles and have no validation error.
+        for (CcdBundleDTO bundle : ccdBundleDtos) {
+            final ArrayNode bundles = ccdCallbackDto.findCaseProperty(ArrayNode.class).orElseGet(() -> {
                 ArrayNode arrayNode = jsonMapper.createArrayNode();
-                ((ObjectNode)ccdCallbackDto.getCaseData()).set(ccdCallbackDto.getPropertyName().get(), arrayNode);
+                ((ObjectNode) ccdCallbackDto.getCaseData()).set(ccdCallbackDto.getPropertyName().get(), arrayNode);
                 return arrayNode;
             });
 
-        CcdBundleDTO bundle = bundleFactory.create(configuration, ccdCallbackDto.getCaseData());
-        ccdCallbackDto.setEnableEmailNotification(bundle.getEnableEmailNotificationAsBoolean());
-        if (bundle.getFileNameIdentifier() != null && !bundle.getFileNameIdentifier().isEmpty()) {
-            bundle.setFileName(ccdCallbackDto.getIdentifierFromCcdPayload(bundle.getFileNameIdentifier()) + "-" + bundle.getFileName());
-        }
-        bundle.setCoverpageTemplateData(ccdCallbackDto.getCaseDetails());
-
-        automatedStitchingExecutor.startStitching(
+            automatedStitchingExecutor.startStitching(
                 ccdCallbackDto.getCaseId(),
                 ccdCallbackDto.getJwt(),
                 bundle);
 
-        bundles.insert(0, bundleDtoToBundleJson(bundle));
+            bundles.insert(0, bundleDtoToBundleJson(bundle));
+        }
 
         return ccdCallbackDto.getCaseData();
     }
 
+    private List<CcdBundleDTO> populateBundleConfigs(CcdCallbackDto ccdCallbackDto, List<String> bundleConfigurations) {
+        List<CcdBundleDTO>  ccdBundleDtos = new ArrayList<>();
+        for (String bundleConfig : bundleConfigurations) {
+
+            BundleConfiguration configuration = configurationLoader.load(bundleConfig);
+
+
+            CcdBundleDTO bundle = bundleFactory.create(configuration, ccdCallbackDto.getCaseData());
+            ccdCallbackDto.setEnableEmailNotification(bundle.getEnableEmailNotificationAsBoolean());
+            if (StringUtils.isNotBlank(bundle.getFileNameIdentifier())) {
+                bundle.setFileName(ccdCallbackDto.getIdentifierFromCcdPayload(bundle.getFileNameIdentifier()) + "-" + bundle.getFileName());
+            }
+            bundle.setCoverpageTemplateData(ccdCallbackDto.getCaseDetails());
+
+            //            Set<ConstraintViolation<CcdBundleDTO>> violations = validator.validate(bundle);
```

Comment:
since this is not needed for now, let's remove it, and we can always get it back from Git